### PR TITLE
DEPR: deprecate sortlevel in favor of sort_index

### LINKED ISSUE #14279: DEPR: sortlevel
when `.sort` and co were changed to be `.sort_values`, `sort_index`. IIRC this was changed to make the change less disruptive. This can be directly replaced by `sort_index`.

### PR DESCRIPTION
**Pull request by jreback**
closes #14279

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3360,6 +3362,8 @@ def sortlevel(self, level=0, axis=0, ascending=True, inplace=False,
         DataFrame.sort_index(level=...)
 
         """
+        warnings.warn("sortlevel is deprecated, use sort_index(...)",
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: maybe 'sort_index(level=...)' to be a bit more specific?

File: pandas/core/series.py [hunk 2]
```diff
@@ -2006,6 +2008,8 @@ def sortlevel(self, level=0, ascending=True, sort_remaining=True):
         Series.sort_index(level=...)
 
         """
+        warnings.warn("sortlevel is deprecated, use sort_index(...)",
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]: same here

**Review by jorisvandenbossche — APPROVED** [review]
Great!

Minor comment.

Apart from that, should we add a basic sortlevel test that tests that a warning is raised and that it gives the same as sort_index?

**Comment by jreback** [comment]
updated

**Pull request merged**